corim/meta.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package corim
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/veraison/corim/comid"
+)
+
+type Signer struct {
+	Name string           `cbor:"0,keyasint" json:"name"`
+	URI  *comid.TaggedURI `cbor:"1,keyasint,omitempty" json:"uri,omitempty"`
+}
+
+func NewSigner() *Signer {
+	return &Signer{}
+}
+
+// SetName sets the given name in Signer
+func (o *Signer) SetName(name string) *Signer {
+	if o != nil {
+		if name == "" {
+			return nil
+		}
+		o.Name = name
+	}
+	return o
+}
+
+// SetURI sets the given uri in Signer
+func (o *Signer) SetURI(uri string) *Signer {
+	if o != nil {
+		if uri == "" {
+			return nil
+		}
+
+		taggedURI, err := comid.String2URI(&uri)
+		if err != nil {
+			return nil
+		}
+
+		o.URI = taggedURI
+	}
+	return o
+}
+
+// Valid checks the validity of individual fields within Signer
+func (o Signer) Valid() error {
+	if o.Name == "" {
+		return errors.New("empty name")
+	}
+
+	if o.URI != nil {
```

Comment:
Should we check if it is a valid URI instead of just checking for a empty URI ?

Reply:
good point!

Reply:
done in 122140e